projects/week_2/Instructions.md

```diff
@@ -0,0 +1,275 @@
+
+When doing operations on the color data of our images, the color space we use is important. Different color spaces can make different operations on the colors easier or harder. The RGB image space you're probably most familiar with is great for storing images that will be displayed with a digitial screen. RGB breaks our colors into red, green, and blue components. This is convenient as most displays use red, green, and blue lights in each pixel to render the image. This is not a very helpful color space for color detection though, because it can be complicated to define the volume occupied by a single, conceptual color.
+
+A color space that's easier to work with for color detection is "HSV". This color space still uses three channels, but now these channels represent the hue, saturation, and value properties of the color. Hue tells us the dominant frequency of the light, or the part of light that we most associate with color. Hue is a cylcic value often associated with angles. Red lies at the wrap-around point (0 or 360 degrees). Saturation tells us how "pure" the color is, with low saturation being completely grayscale, and high saturation being the pure color. Value tells us the brightness of the color. Low values are dark colors, with 0 value being black.
+
+This color space isn't perfect. Note that there are certain kinds of singularities where multiple HSV values can be used to represent the exact same color, such as white or black. In practice, however, it's very useful for detecting most colors.
+
+![The HSV cylinder](https://upload.wikimedia.org/wikipedia/commons/thumb/3/33/HSV_color_solid_cylinder_saturation_gray.png/320px-HSV_color_solid_cylinder_saturation_gray.png)
+
+Thus, our implementation of `FindColors` will use two steps: convert our input image to the HSV color space then populate a single-channel output image based on the HSV value at each pixel. We'll start by opening up [student_functions.cpp](../../obstacle_detector/src/student_functions.cpp) and adding an empty definition for `FindColors`:
+
+```c++
+cv::Mat FindColors(const cv::Mat input, const cv::Scalar range_min, const cv::Scalar range_max)
+{
+}
+```
+
+Now declare a variable to hold the HSV version of our input image. This should be a `cv::Mat` variable called `input_hsv`. Then, call the `cvtColor` function from OpenCV to do the conversion to HSV:
+
+```c++
+cv::Mat input_hsv;
+cv::cvtColor(input, input_hsv, cv::COLOR_BGR2HSV);
+```
+
+Note that the conversion type uses "BGR". This color space uses the same channels as RGB, just in the opposite order in the image data. OpenCV uses BGR as the default color space for color images.
+
+Next, declare another `cv::Mat` variable that will hold our output image, named `output`. We'll initialize this variable to be the same size as our input image while holding one channel of unsigned, 8-bit values.
```

Comment:
Do we talk about what an unsigned bit is in the C++ videos? I know we don't in the robotics side.

Reply:
Yes, they get mentioned in the C++ videos